sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -914,19 +914,19 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
       selectExpressions: Seq[NamedExpression],
       query: LogicalPlan): LogicalPlan = withOrigin(ctx) {
     if (ctx.groupingExpressionsWithGroupingAnalytics.isEmpty) {
-      val groupByExpressions = expressionList(ctx.groupingExpressions)
       if (ctx.GROUPING != null) {
-        // GROUP BY .... GROUPING SETS (...)
-        // `GROUP BY warehouse, product GROUPING SETS((warehouse, producets), (warehouse))` is
-        // semantically equivalent to `GROUP BY GROUPING SETS((warehouse, produce), (warehouse))`.
-        // Under this grammar, the fields appearing in `GROUPING SETS`'s groupingSets must be a
-        // subset of the columns appearing in group by expression.
+        // GROUP BY GROUPING SETS (...)
+        if (!ctx.groupingExpressions.isEmpty) {
```

Comment:
We cannot use a legacy option to deprecate this syntax?

Reply:
It's just a warning message, and not worth a config to turn it off.

Reply:
failing is a bit overkill and we can consider it after one release probably.

Reply:
> We cannot use a legacy option to deprecate this syntax?

Since we have no behavior change, not need to add legacy option. and @cloud-fan Do we need to add this in SQL migration guide or other place to tell end user not to use this?

Since many use case real user won't see this warn message.

Reply:
It's not a breaking change so usually we don't put it in the migration guide.

Reply:
Is it possible to show the warning log only once? I'm a bit worried that existing  users operate a service that use the syntax in production and they will receive many warning messages when upgrading Spark into 3.2.0+. Or, `logWarning` -> `logInfo`?

Reply:
I've removed the warning message.